pylint/checkers/__init__.py

```diff
@@ -46,28 +46,38 @@
 
 """
 
+from typing import Counter, Dict, List, Tuple, Union
+
 from pylint.checkers.base_checker import BaseChecker, BaseTokenChecker
 from pylint.checkers.deprecated import DeprecatedMixin
 from pylint.checkers.mapreduce_checker import MapReduceMixin
 from pylint.utils import diff_string, register_plugins
 
 
-def table_lines_from_stats(stats, old_stats, columns):
+def table_lines_from_stats(
+    stats: Dict[
+        str, Union[int, Counter[str], List, Dict[str, Union[int, str, Dict[str, int]]]]
+    ],
+    old_stats: Dict[
+        str, Union[int, Counter[str], List, Dict[str, Union[int, str, Dict[str, int]]]]
+    ],
```

Comment:
I'm a bit surprised that the code works with so many different types.
@Pierre-Sassoulas mentioned that we should probably create a `Stats` class or something like it, I agree. However as it would be a larger refactoring for another time, maybe a TypeAlias will work for now.

I would suggest to create `pylint/typing.py` as a helper file to store TypeAliases, Enums, NamedTuple, TypedDicts, and so on. (Only if it doesn't make sense somewhere else of course.)